mailpoet/lib/API/MP/v1/Subscribers.php

```diff
@@ -260,6 +260,31 @@ public function unsubscribeFromLists($subscriberIdOrEmail, array $listIds): arra
    * @return array
    */
   public function getSubscribers(array $filter, int $limit, int $offset): array {
+    $listingDefinition = $this->buildListingDefinition($filter, $limit, $offset);
+    $subscribers = $this->subscriberListingRepository->getData($listingDefinition);
+    $result = [];
+    foreach ($subscribers as $subscriber) {
+      $result[] = $this->subscribersResponseBuilder->build($subscriber);
+    }
+    return $result;
+  }
+
+  /**
+   * @param array $filter {
+   *     Optional. Filters to retrieve subscribers.
+   *
+   *     @type string        $status       One of values: subscribed, unconfirmed, unsubscribed, inactive, bounced
+   *     @type int           $listId       id of a list or dynamic segment
+   *     @type \DateTime|int $minUpdatedAt DateTime object or timestamp of last update of subscriber.
+   * }
+   * @return int
+   */
+  public function getSubscribersCount(array $filter): int {
```

Comment:
I am not sure what you mean here.

If your idea is to pass the whole filter argument to the listing definition, that is not possible because the API uses `status` as a filter, but the listing uses it as a group. Also, we use internal `segment`, and the API expects `listId`

Reply:
For future reference: We've decided to leave it for now as currently, the number of filters is very limited, later on when we add more filter, it'll make more sense to pass the filters array in form of a class instance instead of an array.

Reply:
@samnajian, @lysyjan  I just want to add here that the $filter as the array was inspired by [get_posts](https://github.com/WordPress/WordPress/blob/c11ce53dceee9d8af402f16af23ab4aa61e6a074/wp-includes/post.php#L2380) and also a few older methods like addSubscriber accepts arrays.
Having it as a class has some advantages, like we can enforce types of parameters using type hints in setters.

> later on when we add more filter, it'll make more sense to pass the filters array in form of a class instance instead of an array.

Switching from array to class would be a BC break, and we will still need to support both array and class. The two new methods `getSubscribers` and `getSubscribersCount` were not released yet, so we still can make a decision. I don't have super strong opinion here. Array is used in WP and in older methods in our API, the class is better for types. So it is kind of alignment with WP and consistency VS types checking.

Reply:
Thanks for your input @costasovo especially since I didn't check if these methods are released or not yet, I personally like the advantage of type-checking as it means a lot of issues can be prevented just by my cold IDE, but that's also very opinionated when it gets to PHP. To me, an array is more suited for pure data, which doesn't carry logic or meaning, like a POJO and objects/class instances for anything that carries logic with itself.  I'll still leave this decision to @lysyjan since we can always refactor this to use class instances when the need is more vivid.